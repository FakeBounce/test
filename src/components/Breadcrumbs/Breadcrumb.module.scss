.Breadcrumb > nav > ol  {
  background-color: transparent;
  padding-left: 0;
  margin-bottom: 0;
}

.Breadcrumb > nav > ol > span,
.Breadcrumb > nav > ol > a {
  font-size: 11px;
  color: #1F2B3E;
  font-weight: 300;
}