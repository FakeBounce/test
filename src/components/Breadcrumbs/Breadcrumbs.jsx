import React, { Component } from "react";
import { Breadcrumb, BreadcrumbItem } from "reactstrap";
import styles from "./Breadcrumb.module.scss";
import { translate } from "common/methods/translations";

class Breadcrumbs extends Component {
  render() {
    const { elements } = this.props;
    return (
      <div className={styles.Breadcrumb}>
        <Breadcrumb tag="nav">
          {elements &&
            elements.map((element, index) => {
              return (
                <BreadcrumbItem
                  key={index}
                  href={element.href}
                  active={element.isActive}
                  tag={element.isActive ? "span" : "a"}
                >
                  {translate(`globals.menu.${element.name}`)}
                </BreadcrumbItem>
              );
            })}
        </Breadcrumb>
      </div>
    );
  }
}

export default Breadcrumbs;
