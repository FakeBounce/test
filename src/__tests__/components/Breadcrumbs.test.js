import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import Breadcrumbs from "components/Breadcrumbs/Breadcrumbs";

configure({ adapter: new Adapter() });

const breadcrumbElements = [
  { name: "home", href: "/home" },
  { name: "myProfile", isActive: true }
];

describe("Testing Breadcrumbs component", () => {
  it("renders as expected without parameters", () => {
    const wrapper = shallow(<Breadcrumbs />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders as expected with parameters", () => {
    const wrapper = shallow(<Breadcrumbs elements={breadcrumbElements} />);
    expect(wrapper).toMatchSnapshot();
  });
});
