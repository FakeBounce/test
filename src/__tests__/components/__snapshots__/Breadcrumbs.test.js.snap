// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Breadcrumbs component renders as expected with parameters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Breadcrumbs
    elements={
        Array [
            Object {
              "href": "/home",
              "name": "home",
            },
            Object {
              "isActive": true,
              "name": "myProfile",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Breadcrumb
        aria-label="breadcrumb"
        listTag="ol"
        tag="nav"
>
        <BreadcrumbItem
                active={undefined}
                href="/home"
                tag="a"
        >
                Accueil
        </BreadcrumbItem>
        <BreadcrumbItem
                active={true}
                href={undefined}
                tag="span"
        >
                Mon Profil
        </BreadcrumbItem>
</Breadcrumb>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-label": "breadcrumb",
        "children": Array [
          <BreadcrumbItem
            active={undefined}
            href="/home"
            tag="a"
>
            Accueil
</BreadcrumbItem>,
          <BreadcrumbItem
            active={true}
            href={undefined}
            tag="span"
>
            Mon Profil
</BreadcrumbItem>,
        ],
        "listTag": "ol",
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "active": undefined,
            "children": "Accueil",
            "href": "/home",
            "tag": "a",
          },
          "ref": null,
          "rendered": "Accueil",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "active": true,
            "children": "Mon Profil",
            "href": undefined,
            "tag": "span",
          },
          "ref": null,
          "rendered": "Mon Profil",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Breadcrumb
          aria-label="breadcrumb"
          listTag="ol"
          tag="nav"
>
          <BreadcrumbItem
                    active={undefined}
                    href="/home"
                    tag="a"
          >
                    Accueil
          </BreadcrumbItem>
          <BreadcrumbItem
                    active={true}
                    href={undefined}
                    tag="span"
          >
                    Mon Profil
          </BreadcrumbItem>
</Breadcrumb>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "aria-label": "breadcrumb",
          "children": Array [
            <BreadcrumbItem
              active={undefined}
              href="/home"
              tag="a"
>
              Accueil
</BreadcrumbItem>,
            <BreadcrumbItem
              active={true}
              href={undefined}
              tag="span"
>
              Mon Profil
</BreadcrumbItem>,
          ],
          "listTag": "ol",
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "active": undefined,
              "children": "Accueil",
              "href": "/home",
              "tag": "a",
            },
            "ref": null,
            "rendered": "Accueil",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "active": true,
              "children": "Mon Profil",
              "href": undefined,
              "tag": "span",
            },
            "ref": null,
            "rendered": "Mon Profil",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing Breadcrumbs component renders as expected without parameters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Breadcrumbs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Breadcrumb
        aria-label="breadcrumb"
        listTag="ol"
        tag="nav"
/>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-label": "breadcrumb",
        "children": undefined,
        "listTag": "ol",
        "tag": "nav",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Breadcrumb
          aria-label="breadcrumb"
          listTag="ol"
          tag="nav"
/>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "aria-label": "breadcrumb",
          "children": undefined,
          "listTag": "ol",
          "tag": "nav",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
